Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,366 @@
+using Content.Shared.Interaction;
+using Content.Shared.Popups;
+using Robust.Server.Maps;
+using Robust.Shared.Configuration;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Content.Server.Salvage
+{
+    public class SalvageSystem : EntitySystem
+    {
+        [Dependency] private readonly IChatManager _chatManager = default!;
+        [Dependency] private readonly IPauseManager _pauseManager = default!;
+        [Dependency] private readonly IMapLoader _mapLoader = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+        private static readonly TimeSpan AttachingTime = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan HoldTime = TimeSpan.FromMinutes(4);
+        private static readonly TimeSpan DetachingTime = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan CooldownTime = TimeSpan.FromMinutes(1);
+
+        private readonly Dictionary<GridId, SalvageGridState> _salvageGridStates = new();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SalvageMagnetComponent, InteractHandEvent>(OnInteractHand);
+            SubscribeLocalEvent<SalvageMagnetComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<SalvageMagnetComponent, ComponentShutdown>(OnMagnetRemoval);
+            SubscribeLocalEvent<GridRemovalEvent>(OnGridRemoval);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundEnd);
+        }
+
+        private void OnRoundEnd(RoundRestartCleanupEvent ev)
+        {
+            _salvageGridStates.Clear();
+        }
+
+        private void OnGridRemoval(GridRemovalEvent ev)
+        {
+            _salvageGridStates.Remove(ev.GridId);
+            foreach(var gridState in _salvageGridStates)
+            {
+                foreach(var magnet in gridState.Value.ActiveMagnets)
+                {
+                    if (magnet.AttachedEntity == ev.EntityUid)
+                    {
+                        magnet.AttachedEntity = null;
+                        magnet.MagnetState = MagnetState.Inactive;
+                        Report("salvage-system-announcement-lost");
+                        return;
+                    }
+                }
+            }
+        }
```

Comment:
On round restart the salvage messages go out + it plays the salvage sound at the start of the new round.

I'd just check deletion in the update loop TBH because these events go out on round restarts too.

Reply:
I can't determine the reason the deletion happened from the update loop. I found an alternate solution.